Tests/Marketplace/marketplace_services.py

```diff
@@ -1567,6 +1651,179 @@ def prepare_release_notes(self, index_folder_path, build_number,
+                extracted_names_from_rn = SPECIAL_DISPLAY_NAMES_PATTERN.findall(rn_entry)
+
+                for name in extracted_names_from_rn:
+                    if name not in display_names:
+                        rn_entry = rn_entry.replace(f'- **{name}**\n', '')
+
+            filtered_entries[display_name] = rn_entry
+
+        return filtered_entries
+
+    @staticmethod
+    def filter_headers_without_entries(release_notes_dict: dict):
+        """
+        Filters out the entity type/name headers if their entries were filtered by tags.
+
+        Args:
+            release_notes_dict (dict): The release notes in a dict object.
+
+        Returns:
+            (dict) A new release notes dict after filtering.
+        """
+        new_release_notes_dict: dict = {}
+        for entity_header, entity_entry in release_notes_dict.items():
+
+            new_entity_entry = {name: entry for name, entry in entity_entry.items()
+                                if entry not in ['', '\n']}
+
+            if new_entity_entry:
+                new_release_notes_dict[entity_header] = new_entity_entry
+
+        return new_release_notes_dict
+
+    @staticmethod
+    def is_release_notes_not_for_entities(release_notes_str, release_notes_dict):
+        """
+        If the release notes didn't formatted into a dict it's probably because one of the following:
+        - In case it's a first release of the pack then the release notes is taken from the pack description,
+        - If it's just an important message for the customers who uses the pack.
+        In both cases the RN entries will probably not contain the entity headers as in our templates.
+
+        Args:
+            release_notes_str (str): The release notes in string.
+            release_notes_dict (dict): The release notes in dict object.
+
+        Returns:
+            (bool) Whther the dict contains the RN entries by the entities types.
+        """
+        return release_notes_str and not release_notes_dict
+
+    def filter_release_notes_by_tags(self, release_notes, marketplace):
+        """
+        Filters out from release notes the sub-entries that are wrapped by tags.
+
+        Args:
+            release_notes(str): The release notes entry.
+
+        Return:
+            (str) The release notes entry after filtering.
+        """
+
+        def remove_tags_section_from_rn(release_notes, marketplace, upload_marketplace):
+
+            start_tag, end_tag = TAGS_BY_MP[marketplace]
+            if start_tag in release_notes and end_tag in release_notes and marketplace != upload_marketplace:
+                logging.info(f"Filtering irrelevant release notes by tags {start_tag}-{end_tag} of marketplace {marketplace} "
+                             f"for pack {self._pack_name} when uploading to marketplace {upload_marketplace}.")
+                return re.sub(fr'{start_tag}{TAGS_SECTION_PATTERN}{end_tag}[\n]*', '', release_notes)
```

Comment:
If this line is executed many times, consider compiling the regex once before (currently it compiles it on every call to this function)